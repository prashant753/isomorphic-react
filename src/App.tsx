import * as React from 'react';
import { BrowserRouter } from 'react-router-dom';
import { createBrowserHistory } from 'history';
import { Provider } from 'react-redux';
import store from './state_management/store';
import Navigation from './navigation/Navigation'

const history = createBrowserHistory();

class App extends React.Component<any> {

  public render() {
    return (
        <Provider store={store}>
        <BrowserRouter>
          <Navigation history={history} />
        </BrowserRouter>
      </Provider>
    )
  }
}

export default App;
