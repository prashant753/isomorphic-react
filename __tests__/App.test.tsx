import React from 'react';
import App from '../src/App';
import { shallow, ShallowWrapper } from 'enzyme';

const createTestProps = (prop: any) => ({
    ...prop,
});
let props: any;
let wrapper: ShallowWrapper;

describe('App', () => {

    beforeEach(() => {
        props = createTestProps({
            homepage: true,
        });
    });
    it('should render App', () => {
        wrapper = shallow(<App {...props} />);
        expect(wrapper).toMatchSnapshot();
    });
});
