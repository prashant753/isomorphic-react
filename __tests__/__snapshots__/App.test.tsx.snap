// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render App 1`] = `
<Provider
  store={
    Object {
      "@@observable": [Function],
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <BrowserRouter>
    <withRouter(Navigation)
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "createHref": [Function],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "pathname": "blank",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    />
  </BrowserRouter>
</Provider>
`;
